docs/how-to/configure-revision-history.md

```diff
@@ -0,0 +1,25 @@
+# How to configure `revision-history-limit`
+
+You can limit how many revisions of the images are kept in OpenStack Glance by specifying the
+`revision-history-limit` configuration option. By default, 5 most recent images are kept.
+
+```
+REVISION_HISTORY_LIMIT=2
+OPENSTACK_AUTH_URL=<openstack-auth-url, e.g. http://my-openstack-deployment/openstack-keystone>
+OPENSTACK_PASSWORD=<openstack project password>
+OPENSTACK_PROJECT_DOMAIN_NAME=<openstack project domain name>
+OPENSTACK_PROJECT_NAME=<openstack project name>
+OPENSTACK_USER_DOMAIN_NAME=<openstack user domain name>
+OPENSTACK_USERNAME=<openstack username>
+
+juju deploy github-runner-image-builder \
+--config revision-history-limit=$REVISION_HISTORY_LIMIT
+--config openstack-auth-url=$OPENSTACK_AUTH_URL \
+--config openstack-password=$OPENSTACK_PASSWORD \
+--config openstack-project-domain-name=$OPENSTACK_PROJECT_DOMAIN_NAME \
+--config openstack-project-name=$OPENSTACK_PROJECT_NAME \
+--config openstack-user-domain-name=$OPENSTACK_USER_DOMAIN_NAME \
+--config openstack-user-name=$OPENSTACK_USERNAME
+```
+
+The example above would keep the two most recent revisions of the image before deletion.
```

Comment:
Unrelated to the howto: Could there be a race condition where a github runner charm references deleted images?

Reply:
If the runners are already spawned with the image, it is okay for the image to be deleted (since the images get loaded already). In the case where it tries to boot and the image is deleted, there should be enough `keep_revisions` parameter set for it to be able to keep on using it.
In the case where `keep_revisions` is low (keep only last 2 images for example), it will soon be patched via relation data update because the deletion happens right before propagation of new image ids. So the downtime due to race condition can happen but it will either be 
1. very low chance of it happening
2. very low downtime because of it